src/components/safe-messages/MsgModal/index.tsx

```diff
@@ -93,6 +97,40 @@ const MsgModal = ({
 
   const isDisabled = !isOwner || hasSigned || !onboard
 
+  const [isPolling, setIsPolling] = useState(false)
+
+  // TODO: move to hook
+  const [ongoingMessage, setOngoingMessage] = useState<Omit<SafeMessage, 'type'>>()
+  const CONFIRMATIONS_POLLING_INTERVAL = 3000
+  useEffect(() => {
+    if (!isPolling) return
+
+    const fetchSafeMessage = async () => {
+      try {
+        const message = await getSafeMessage(safe.chainId, safeMessageHash)
+        setOngoingMessage(message)
+
+        if (message.confirmationsSubmitted === message.confirmationsRequired) {
+          setIsPolling(false)
+          clearInterval(intervalId)
+        }
+
+        console.log('message', message)
```

Comment:
Here we should emit the `SafeMsgEvent.SIGNATURE_PREPARED` event with the `requestID` and `signature` and listen to that in the `AppFrame`